### PR TITLE
Add Bundle Announcement + Refactor What's New Menu CTA

### PR DESCRIPTION
**Pull request by codemist**
# New feature description

This adds the Bundle announcement on the Whats New Menu as well as refactors logic of the upgrade button within an Whats New item 

# Screenshot (if applicable)

image
image

# How to test

Check that the Bundle item is present in the What's New Menu. 

# Checklist

- [x] l10n changes have been submitted to the l10n repository, if any.
- [x] All acceptance criteria are met.
- [x] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (codemist) [inline frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:469]: `<CtaLinkButton>` determines whether to show an upgrade CTA to a user, by checking if they already possess the plan in question. It can also be reused in future What's New entries with a upgrade CTA, like the previous Phone masking and Bundle announcements.

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (codemist) [inline frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:514]: Waiting on a `profile.has_bundle` object to be built out, similar to `has_phone` so that we do not need to show the upgrade CTA to users who already have that plan.

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (codemist) [inline frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:500]: Still waiting on a confirmation of copy.

File: frontend/pendingTranslations.ftl
Comment (Vinnl) [inline frontend/pendingTranslations.ftl:404]:
```suggestion
#   $savings (string) - the percentage saved (including % symbol) for a given plan. Examples: 50%, 70%
```

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (Vinnl) [inline frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:468]: To avoid having to choose a specific billing period for the user here, [Eduardo suggested](https://mozilla.slack.com/archives/C013CSYEL5T/p1662565686526129?thread_ts=1662548886.744219&cid=C013CSYEL5T) making this link point to `/premium#pricing` instead. (Which will need [a Next.js `<Link>`](https://nextjs.org/docs/routing/introduction#linking-between-pages) instead of an `<OutboundLink>`.)

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (Vinnl) [reply frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:500]: In case you do get the final percentage before merging this, could you also insert that same value [here](https://github.com/mozilla/fx-private-relay/blob/8e6df3631c2217c0b040c7127d96228c8f1abb71/frontend/src/components/landing/PlanMatrix.tsx#L468) and [here](https://github.com/mozilla/fx-private-relay/blob/8e6df3631c2217c0b040c7127d96228c8f1abb71/frontend/src/components/landing/PlanMatrix.tsx#L298)?

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (Vinnl) [reply frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:514]: Could callout! Asked about that [on Slack](https://mozilla.slack.com/archives/C013CSYEL5T/p1663062840438759), since I'm not sure whether that's on anybody's radar.

**Review by Vinnl — APPROVED** [review]
Looking good!

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (codemist) [reply frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:514]: Thank you!

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (codemist) [reply frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:500]: Will do. I'll ask Tony if these values are confirmed.

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (codemist) [inline frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:109]: Change the link to `/premium#pricing` as opposed to straight to Subplat. I left the `link` prop as an option we may use in the future.

**Review by Vinnl — APPROVED** [review]

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (codemist) [reply frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx]: done.

File: frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx
Comment (Vinnl) [reply frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:507]: OK, submitted https://github.com/mozilla/fx-private-relay/pull/2505, so we can add a check for `profile.has_premium && profile.has_vpn` in here (or possibly just the latter?).

Comment (codemist) [reply frontend/src/components/layout/navigation/whatsnew/WhatsNewMenu.tsx:507]: Awesome, I'll add a check for `profile.has_vpn`

**Pull request merged**
